internal/metrics/logfilemetricexporter/factory_test.go

```diff
@@ -191,6 +191,39 @@ var _ = Describe("LogFileMetricExporter functions", func() {
 			Expect(podSpec.Tolerations).To(ContainElement(testTol1))
 			Expect(podSpec.Tolerations).To(Equal(finalTolerations))
 		})
-	})
 
+		It("should not have duplicate tolerations in the pod spec", func() {
+			testTol1 := corev1.Toleration{
+				Key:      "node-role.kubernetes.io/master",
+				Operator: corev1.TolerationOpEqual,
+				Effect:   corev1.TaintEffectNoSchedule,
+			}
+
+			logFileMetricExporter.Spec.Tolerations = []corev1.Toleration{
+				testTol1,
+			}
+
+			finalTolerations := []corev1.Toleration{
+				{
+					Key:      "node-role.kubernetes.io/master",
+					Operator: corev1.TolerationOpEqual,
+					Effect:   corev1.TaintEffectNoSchedule,
+				},
+				{
+					Key:      "node.kubernetes.io/disk-pressure",
+					Operator: corev1.TolerationOpExists,
+					Effect:   corev1.TaintEffectNoSchedule,
+				},
+			}
```

Comment:
It would be clearer if testTol1 was included into finalTolerations as well so it's clear it is the same toleration.